packages/@webex/media-helpers/test/unit/spec/webrtc-core.js

```diff
@@ -1,45 +1,130 @@
+import {LocalCameraTrack, LocalMicrophoneTrack, LocalMicrophoneTrackEvents, LocalCameraTrackEvents, LocalDisplayTrack, createCameraTrack, createMicrophoneTrack, createDisplayTrack} from '@webex/media-helpers';
+import * as wcmetracks from '@webex/internal-media-core';
 
 describe('media-helpers', () => {
   describe('webrtc-core', () => {
 
     const classesToTest = [
-      {className: LocalCameraTrack, title: 'LocalCameraTrack'},
-      {className: LocalMicrophoneTrack, title: 'LocalMicrophoneTrack'},
+      {className: LocalCameraTrack, title: 'LocalCameraTrack', event: LocalCameraTrackEvents, createFn: createCameraTrack, spyFn: 'createCameraTrack'},
+      {className: LocalMicrophoneTrack, title: 'LocalMicrophoneTrack', event: LocalMicrophoneTrackEvents, createFn: createMicrophoneTrack, spyFn: 'createMicrophoneTrack'},
     ];
 
-    classesToTest.forEach(({className, title}) =>
-      describe(title, () => {
-        const fakeStream = {
-          getTracks: sinon.stub().returns([{
-            label: 'fake track',
-            id: 'fake track id',
-          }])
-        };
-        const track = new className(fakeStream);
+    classesToTest.forEach(({className, title, event, createFn, spyFn}) =>
+    describe(title, () => {
+      const fakeStream = {
+        getTracks: sinon.stub().returns([{
+          label: 'fake track',
+          id: 'fake track id',
+        }])
+      };
+      const track = new className(fakeStream);
 
-        it('by default allows unmuting', async () => {
-          assert.equal(track.isUnmuteAllowed(), true);
-          await track.setMuted(false);
-        })
+      afterEach(() => {
+        sinon.restore();
+      });
 
-        it('rejects setMute(false) if unmute is not allowed', async () => {
-          track.setUnmuteAllowed(false);
+      it('by default allows unmuting', async () => {
+        assert.equal(track.isUnmuteAllowed(), true);
+        await track.setMuted(false);
+      })
 
-          assert.equal(track.isUnmuteAllowed(), false);
-          await assert.isRejected(track.setMuted(false), Error, 'Unmute is not allowed');
-        });
+      it('rejects setMute(false) if unmute is not allowed', async () => {
+        track.setUnmuteAllowed(false);
 
-        it('resolves setMute(false) if unmute is allowed', async () => {
-          track.setUnmuteAllowed(true);
+        assert.equal(track.isUnmuteAllowed(), false);
+        const fn = () => track.setMuted(false);
+        expect(fn).to.throw(/Unmute is not allowed/);
+      });
 
-          assert.equal(track.isUnmuteAllowed(), true);
-          await track.setMuted(false);
-        })
+      it('resolves setMute(false) if unmute is allowed', async () => {
+        track.setUnmuteAllowed(true);
+
+        assert.equal(track.isUnmuteAllowed(), true);
+        track.setMuted(false);
+      });
+
+      it('tests setServerMuted- true to false', async () => {
+        track.underlyingTrack.enabled = true; //start with unmuted
```

Comment:
why do we need this? isn't the call to setMuted() enough to set the right state?

Reply:
Here's  the reason
![image](https://user-images.githubusercontent.com/82164376/232730592-bf9dff27-4f52-4e37-af66-05fd200272a8.png)

Reply:
setMuted calls super.setMuted which as you see in screenshot does something conditionally, hence setMuted isn't doing anything as this condition is not satisfied.